docs/src/rfc/text/0017-aml-parser.md

```diff
@@ -0,0 +1,331 @@
+# RFC: `ACPI SDT AML Handler`
+
+This RFC proposes a Rust-based AML (ACPI Machine Language) service that extends the existing ACPI service by
+providing a safe, ergonomic interface for parsing and modifying AML bytecode in firmware during the DXE phase.
+The implementation mirrors the ACPI SDT protocol’s functionality for AML traversal and patching
+and is designed mainly for firmware use rather than OS-level interpretation.
+It defines a structured system of AML handles for navigating AML streams and a trait-based `AmlParser` service for operations
+such as opening tables, iterating operands, modifying values, and traversing child or sibling nodes.
+The goal is to replace legacy C-based AML handling with a type-safe Rust service that supports ACPI 2.0+.
+Future extensions may include extending this infrastructure for application-side AML interpretation 
+within a `patina-acpi` crate.
+
+## Change Log
+
+- 2025-10-1: Initial RFC created.
+- 2025-10-22: Update iteration interface and address prior art.
+- 2025-10-28: Move to FCP.
+- 2025-10-28: Add notes based on offline discussion of AML buffer parsing.
+
+## Motivation
+
+This RFC is an extension of the [ACPI service](0005-acpi.md).
+Similar to the ACPI service, this Rust-based AML service will provide a safer and
+more ergonomic interface for parsing and modifying AML bytecode.
+
+## Technology Background
+
+AML bytecode is encoded mainly in the body of the DSDT and SSDT.

```

Comment:
While the DSDT is a single table, a system can have zero, one, or multiple SSDTs.

```suggestion
Compiled ACPI tables such as the DSDT and SSDTs are composed of AML bytecode.

```